extensions/resteasy-reactive/quarkus-resteasy-reactive/runtime/src/main/java/io/quarkus/resteasy/reactive/server/runtime/websocket/VertxWebSocketRestHandler.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.resteasy.reactive.server.runtime.websocket;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.jboss.resteasy.reactive.server.core.ResteasyReactiveRequestContext;
+import org.jboss.resteasy.reactive.server.model.HandlerChainCustomizer;
+import org.jboss.resteasy.reactive.server.spi.ServerRestHandler;
+
+import io.vertx.core.Handler;
+import io.vertx.core.http.ServerWebSocket;
+
+public class VertxWebSocketRestHandler implements HandlerChainCustomizer {
+
+    @Override
+    public List<ServerRestHandler> handlers(Phase phase) {
+        if (phase == Phase.AFTER_METHOD_INVOKE) {
+            return Collections.singletonList(new ServerRestHandler() {
+                @Override
+                public void handle(ResteasyReactiveRequestContext requestContext) throws Exception {
+                    for (Object i : requestContext.getParameters()) {
+                        if (i instanceof ServerWebSocket) {
+                            ServerWebSocket socket = (ServerWebSocket) i;
+                            socket.closeHandler(new Handler<Void>() {
+                                @Override
+                                public void handle(Void event) {
+                                    requestContext.close();
+                                }
+                            });
+                        }
+                    }
+                    requestContext.suspend(); //we never resume
```

Comment:
Check out what I did in `MultiResponseHandler` which is similar to this, where I suspend, then do this:

```java
            // let's make sure we never restart by accident, also make sure we're not marked as completed
            requestContext.restart(AWOL, true);
```